src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.ts

```diff
@@ -0,0 +1,192 @@
+import { DimlessBinaryPipe } from '~/app/shared/pipes/dimless-binary.pipe';
+
+@Component({
+  selector: 'cd-nvmeof-namespaces-form',
+  templateUrl: './nvmeof-namespaces-form.component.html',
+  styleUrls: ['./nvmeof-namespaces-form.component.scss']
+})
+export class NvmeofNamespacesFormComponent implements OnInit {
+  action: string;
+  permission: Permission;
+  poolPermission: Permission;
+  resource: string;
+  pageURL: string;
+  edit: boolean = false;
+  nsForm: CdFormGroup;
+  subsystemNQN: string;
+  rbdPools: Array<Pool> = null;
+  units: Array<string> = ['KiB', 'MiB', 'GiB', 'TiB'];
+  nsid: string;
+  currentBytes: number;
+  invalidSizeError: boolean;
+
+  constructor(
+    public actionLabels: ActionLabelsI18n,
+    private authStorageService: AuthStorageService,
+    private taskWrapperService: TaskWrapperService,
+    private nvmeofService: NvmeofService,
+    private poolService: PoolService,
+    private rbdService: RbdService,
+    private router: Router,
+    private route: ActivatedRoute,
+    public activeModal: NgbActiveModal,
+    public formatterService: FormatterService,
+    public dimlessBinaryPipe: DimlessBinaryPipe
+  ) {
+    this.permission = this.authStorageService.getPermissions().nvmeof;
+    this.poolPermission = this.authStorageService.getPermissions().pool;
+    this.resource = $localize`Namespace`;
+    this.pageURL = 'block/nvmeof/subsystems';
+  }
+
+  init() {
+    this.createForm();
+    this.action = this.actionLabels.CREATE;
+    this.route.params.subscribe((params: { subsystem_nqn: string; nsid: string }) => {
+      this.subsystemNQN = params.subsystem_nqn;
+      this.nsid = params?.nsid;
+    });
+  }
+
+  initForEdit() {
+    this.edit = true;
+    this.action = this.actionLabels.EDIT;
+    this.nvmeofService
+      .getNamespace(this.subsystemNQN, this.nsid)
+      .subscribe((res: NvmeofSubsystemNamespace) => {
+        const convertedSize = this.dimlessBinaryPipe.transform(res.rbd_image_size).split(' ');
+        this.currentBytes = res.rbd_image_size;
+        this.nsForm.get('image').setValue(res.rbd_image_name);
+        this.nsForm.get('pool').setValue(res.rbd_pool_name);
+        this.nsForm.get('unit').setValue(convertedSize[1]);
+        this.nsForm.get('image_size').setValue(convertedSize[0]);
+        this.nsForm.get('image_size').addValidators(Validators.required);
+        this.nsForm.get('image').disable();
+        this.nsForm.get('pool').disable();
+      });
+  }
+
+  initForCreate() {
+    this.poolService.getList().subscribe((resp: Pool[]) => {
+      this.rbdPools = resp.filter(this.rbdService.isRBDPool);
+    });
+  }
+
+  ngOnInit() {
+    this.init();
+    if (this.router.url.includes('subsystems/(modal:edit')) {
+      this.initForEdit();
+    } else {
+      this.initForCreate();
+    }
+  }
+
+  createForm() {
+    this.nsForm = new CdFormGroup({
+      image: new UntypedFormControl(`image:${Date.now()}`, {
```

Comment:
to better distinguish it?
```suggestion
      image: new UntypedFormControl(`nvme_ns_image:${Date.now()}`, {
```